### PR TITLE
Add 'Intro to Notion API' example

### PR DESCRIPTION
**Pull request by jessmitch42**
Adds a new example to the `/examples` directory. There was a request to have more code samples that build up in complexity so each section has a simple version that gets progressively more complex. 
There will be an `advanced` section added in a future PR (likely some existing examples moved over here to organize them a bit more). 
This can be easily expanded as more requests come in; for example, the current `/basic` files can be grouped into a `blocks` subdirectory if other "basic" examples get requested.

Note: Part of the request was to heavily comment this code here.

### CODE REVIEW

**Review by parulbaweja — DISMISSED** [review]
Just a couple questions, but lgtm!

File: examples/intro-to-notion-api/intermediate/3-query-database.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  if (!databaseId) return
+
+  console.log("Adding new pages...")
+  for (let i = 0; i < propertiesForNewPages.length; i++) {
+    // Add a few new pages to the database that was just created
+    await addNotionPageToDatabase(databaseId, propertiesForNewPages[i])
+  }
+
+  // After adding pages, query the database entries (pages)
+  queryDatabase(databaseId)
```
[the remark this answers is not in the document]
Comment (jessmitch42) [reply examples/intro-to-notion-api/intermediate/3-query-database.js:99]: sure thing, just added :)

File: examples/intro-to-notion-api/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+NOTION_API_KEY=<your-notion-api-key>
+NOTION_PAGE_ID=<notion-page-id>
+```
+
+`NOTION_API_KEY`: Create a new integration in the [integrations dashboard](https://www.notion.com/my-integrations) and retrieve the API key from the integration's `Secrets` page.
+
+`NOTION_PAGE_ID`: Use the ID of any Notion page that you want to test adding content to.
+
+The page ID is the 32 character string at the end of any page URL.
+![A Notion page URL with the ID highlighted](./assets/page_id.png)
```
[the remark this answers is not in the document]
Comment (jessmitch42) [reply examples/intro-to-notion-api/README.md:64]: it is!

**Comment by jessmitch42** [comment]
Hey @parulbaweja thanks for the review. I added your suggestion so it looks like it needs another review. If everything looks good, do you mind merging it too after approving? I don't have permissions to merge in this repo 🥲

**Review by parulbaweja — APPROVED** [review]

**Pull request merged**